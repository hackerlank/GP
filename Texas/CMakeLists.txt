
set(showhand_path
    "${CMAKE_HOME_DIRECTORY}/ShowhandAndTexasGame/Showhand"
)

set(texas_path
    "${CMAKE_HOME_DIRECTORY}/ShowhandAndTexasGame/Texas"
)

#add_subdirectory(${showhand_path})
add_subdirectory(${texas_path})