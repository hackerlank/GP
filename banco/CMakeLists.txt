
project(BaccaratGame)

set(baccarat_game_src_path
	"${CMAKE_HOME_DIRECTORY}/BaccaratGame/"
)

set(baccarat_game_out_path
	"${CMAKE_HOME_DIRECTORY}/Output/BaccaratGame/"
)

option(BUILD_BACCARAT_HUNDRED_ALLOC "build baccarat hundred alloc server" TRUE)
if (BUILD_BACCARAT_HUNDRED_ALLOC)
	# add alloc server
	set(alloc_src_path
		"${baccarat_game_src_path}/AllocServer/many"
	)
	add_subdirectory(${alloc_src_path})
endif()

option(BUILD_BACCARAT_ALLOC "build baccarat alloc server" TRUE)
if (BUILD_BACCARAT_ALLOC)
	# add alloc server
	set(alloc_src_path
			"${baccarat_game_src_path}/AllocServer/trunk"
			)
	add_subdirectory(${alloc_src_path})
endif()

option(BUILD_BACCARAT_GAME "build baccarat game server" TRUE)
if (BUILD_BACCARAT_GAME)
	# add game server
	set(hundred_src_path
		"${baccarat_game_src_path}/GameServer/trunk/src"
	)
	add_subdirectory(${hundred_src_path})
endif()

option(BUILD_BACCARAT_HANDRED_GAME "build baccarat hundred game server" TRUE)
if (BUILD_BACCARAT_HANDRED_GAME)
	# add game server
	set(hundred_src_path
			"${baccarat_game_src_path}/GameServer/many/src"
			)
	add_subdirectory(${hundred_src_path})
endif()

option(BUILD_BACCARAT_ROBOT "build baccarat robot server" TRUE)
if (BUILD_BACCARAT_ROBOT)
	# add robot server
	set(hundred_robot_src_path
		"${baccarat_game_src_path}/RobotServer/trunk/src"
	)
	add_subdirectory(${hundred_robot_src_path})
endif()

option(BUILD_BACCARAT_ROBOT_CRYPT "build baccarat robot crypt server" TRUE)
if (BUILD_BACCARAT_ROBOT_CRYPT)
	# add robot server
	set(hundred_robot_crypt_src_path
		"${baccarat_game_src_path}/RobotServerCrypt/"
	)
	add_subdirectory(${hundred_robot_crypt_src_path})
endif()