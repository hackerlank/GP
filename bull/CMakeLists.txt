
project(BullFight2017)

set(bull_game_src_path
	"${CMAKE_HOME_DIRECTORY}/BullFight2017/"
)

set(bull_game_out_path
	"${CMAKE_HOME_DIRECTORY}/Output/BullFight2017/"
)

option(BUILD_BULL2017_ALLOC "build bull alloc server" TRUE)
if (BUILD_BULL2017_ALLOC)
	# add alloc server
	set(alloc_src_path
		"${bull_game_src_path}/AllocServer/"
	)
	add_subdirectory(${alloc_src_path})
endif()

option(BUILD_BULL2017_GAME "build bull hundred game server" TRUE)
if (BUILD_BULL2017_GAME)
	# add hundred server
	set(hundred_src_path
		"${bull_game_src_path}/GameServer/many/src"
	)
	add_subdirectory(${hundred_src_path})
endif()

option(BUILD_BULL2017_ROBOT "build hundred robot server" TRUE)
if (BUILD_BULL2017_ROBOT)
	# add hundred robot server
	set(hundred_robot_src_path
		"${bull_game_src_path}/RobotServer/many/src"
	)
	add_subdirectory(${hundred_robot_src_path})
endif()