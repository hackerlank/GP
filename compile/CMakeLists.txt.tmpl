cmake_minimum_required(VERSION 2.8.6)
project(DD_Games)

set(sdk_path
        ${CMAKE_HOME_DIRECTORY}/DD_SDK
        )

set(crypt_sdk_inc
        ${CMAKE_HOME_DIRECTORY}/DDCrypt_SDK/include
)

set(crypt_sdk_path
        ${CMAKE_HOME_DIRECTORY}/DDCrypt_SDK/LibCentos
        ${CMAKE_HOME_DIRECTORY}/DDCrypt_SDK/openssl/linux_lib
	${CMAKE_HOME_DIRECTORY}/DDCrypt_SDK/third/curl/lib
)
# cmake
set(sdk_cmake_path
        ${sdk_path}/cmake
        )

include(${sdk_cmake_path}/base.cmake)
include(${sdk_cmake_path}/common.cmake)
        
set(crypt_libs
    "${LIB_PRE_NAME}DDCrypt.${LIB_EXT_NAME}"
    "${LIB_PRE_NAME}curl.${LIB_EXT_NAME}"
    "${LIB_PRE_NAME}ssl.${LIB_EXT_NAME}"
    "${LIB_PRE_NAME}crypto.${LIB_EXT_NAME}"
)

