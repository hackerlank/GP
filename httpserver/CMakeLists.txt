IF (__DDCRYPTO__)
project (HttpCSvr)
ELSE ()
project (HttpSvr)
ENDIF ()
cmake_minimum_required(VERSION 2.8.6)
SET(CMAKE_CXX_FLAGS "-std=c++0x -g")

macro(source_group_by_dir source_files strip_head)
    if(CMAKE_SYSTEM MATCHES "Windows" OR CMAKE_SYSTEM MATCHES "Darwin")
        set(sgbd_cur_dir ${CMAKE_CURRENT_SOURCE_DIR})
        foreach(sgbd_file ${${source_files}})
            string(REGEX REPLACE ${sgbd_cur_dir}/\(.*\) \\1 sgbd_fpath ${sgbd_file})
            string(REGEX REPLACE "\(.*\)/.*" \\1 sgbd_group_name ${sgbd_fpath})
            string(COMPARE EQUAL ${sgbd_fpath} ${sgbd_group_name} sgbd_nogroup)
            string(REPLACE "/" "\\" sgbd_group_name ${sgbd_group_name})
            if(sgbd_nogroup)
                set(sgbd_group_name "\\")
            else()
                set(strip_len 0)
                string(LENGTH ${strip_head} strip_len)
                set(group_len 0)
                string(LENGTH ${sgbd_group_name}, group_len)
                string(COMPARE EQUAL ${strip_head} "./" equal_cur)
                if (NOT (${strip_head} STREQUAL "./"))
                    string(SUBSTRING ${sgbd_group_name} ${strip_len} ${group_len} sgbd_group_name)
                endif()
                #message(${sgbd_group_name})
            endif(sgbd_nogroup)
            source_group(${sgbd_group_name} FILES ${sgbd_file})
        endforeach(sgbd_file)
    endif(CMAKE_SYSTEM MATCHES "Windows" OR CMAKE_SYSTEM MATCHES "Darwin")
endmacro(source_group_by_dir)


IF (__DDCRYPTO__)
set(proj_name HttpCSvr)
ELSE ()
set(proj_name HttpSvr)
ENDIF ()

set(src_path
	"./"
)


#list all files
file(GLOB_RECURSE project_headers *.h)
file(GLOB_RECURSE project_hppeaders *.hpp)
file(GLOB_RECURSE handle_cpps http_handle/*.cpp)
file(GLOB_RECURSE comm_cpps comm/*.cpp)
file(GLOB main_cpps *.cpp)
set(all_files ${comm_cpps} ${handle_cpps} ${main_cpps} ${project_headers} ${project_hppeaders})
source_group_by_dir(all_files ${src_path})

set(common_dir ${CMAKE_HOME_DIRECTORY}/comm)
set(third_dir ${CMAKE_HOME_DIRECTORY}/third)
set(mysql_dir ${third_dir}/mysql)
set(libevent_dir ${third_dir}/libevent)
set(xml_dir ${third_dir}/tinyxml)

set(curl_dir ${third_dir}/curl)
set(httpSvrComm_dir ${third_dir}/HttSvrComm)

#add include path
include_directories(${src_path})
include_directories(${common_dir})
include_directories(${common_dir}/Util)

include_directories(${curl_dir}/include)
include_directories(${httpSvrComm_dir}/include)
include_directories(${httpSvrComm_dir}/include/Log)
include_directories(${httpSvrComm_dir}/include/hiredis)

include_directories(${mysql_dir}/include)
include_directories(${libevent_dir}/include)
include_directories(${xml_dir}/include)
#include_directories(/usr/local/include)
link_directories(${mysql_dir}/lib)
link_directories(${libevent_dir}/lib)
link_directories(${xml_dir}/lib)
link_directories(${curl_dir}/lib)
link_directories(${httpSvrComm_dir}/lib)

IF (__DDCRYPTO__)
#cmake -D__DDCRYPTO__=ON ..
add_definitions("-D__DDCRYPTO__")
set(ssl_dir ${third_dir}/openssl)
set(DDCrypt_dir ${third_dir}/DDCrypt)
include_directories(${ssl_dir}/include)
include_directories(${DDCrypt_dir}/include)
link_directories(${ssl_dir}/linux_lib)
link_directories(${DDCrypt_dir}/lib)
ENDIF (__DDCRYPTO__)

add_definitions("-D__STDC_FORMAT_MACROS")
add_definitions("-DCURL_DISABLE_LDAP")

#add exec
add_executable(${proj_name}
	${all_files}
)

#target_link_libraries(${proj_name} -Wl,-Bstatic jsoncpp mysqlclient tinyxml hiredis event crypto DDCrypt -Wl,-Bdynamic pthread z rt)
IF (__DDCRYPTO__)
target_link_libraries(${proj_name} -Wl,-Bstatic HttpSvrComm mysqlclient tinyxml event DDCrypt crypto curl -Wl,-Bdynamic pthread z idn rt dl)
ELSE ()
target_link_libraries(${proj_name} -Wl,-Bstatic HttpSvrComm mysqlclient tinyxml event curl -Wl,-Bdynamic pthread z idn rt)
ENDIF ()