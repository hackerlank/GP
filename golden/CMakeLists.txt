
project(FlowerGame)

set(flower_game_src_path
	"${CMAKE_HOME_DIRECTORY}/FlowerGame/"
)

set(flower_game_out_path
	"${CMAKE_HOME_DIRECTORY}/Output/FlowerGame/"
)

option(BUILD_FLOWER_ALLOC "build flower alloc server" TRUE)
if (BUILD_FLOWER_ALLOC)
	# add alloc server
	set(alloc_src_path
		"${flower_game_src_path}/AllocServer/"
	)
	add_subdirectory(${alloc_src_path})
endif()

option(BUILD_FLOWER_FIGHT_GAME "build flower fight game server" TRUE)
if (BUILD_FLOWER_FIGHT_GAME)
	# add fight server
	set(fight_src_path
		"${flower_game_src_path}/GameServer/fight/src"
	)
	add_subdirectory(${fight_src_path})
endif()

option(BUILD_FLOWER_HUNDRED_GAME "build flower hundred game server" TRUE)
if (BUILD_FLOWER_HUNDRED_GAME)
	# add hundred server
	set(hundred_src_path
		"${flower_game_src_path}/GameServer/hundred/src"
	)
	add_subdirectory(${hundred_src_path})
endif()

option(BUILD_FLOWER_FIGHT_ROBOT "build fight robot server" TRUE)
if (BUILD_FLOWER_FIGHT_ROBOT)
	#set fight src path
	set(figth_robot_src_path
		"${flower_game_src_path}/RobotServer/fight/src"
	)
	add_subdirectory(${figth_robot_src_path})
endif()

option(BUILD_FLOWER_HUNDRED_ROBOT "build hundred robot server" TRUE)
if (BUILD_FLOWER_HUNDRED_ROBOT)
	# add hundred robot server
	set(hundred_robot_src_path
		"${flower_game_src_path}/RobotServer/hundred/src"
	)
	add_subdirectory(${hundred_robot_src_path})
endif()

option(BUILD_FLOWER_HUNDRED_ROBOT_CRYPT "build crypt hundred  robot server" TRUE)
if (BUILD_FLOWER_HUNDRED_ROBOT_CRYPT)
	# add hundred robot server
	set(crpty_hundred_robot_src_path
		"${flower_game_src_path}/RobotServer/many_crypt"
	)
	add_subdirectory(${crpty_hundred_robot_src_path})
endif()

